bootloader/src/pyi_archive.c

```diff
@@ -235,14 +235,15 @@ pyi_arch_extract2fs(ARCHIVE_STATUS *status, TOC *ptoc)
  * The search space uses the given sizes because on Windows and OS X, the code signing
  * will add padding between the end of the COOKIE and the beginning of the signature
  * to align the signature to a quadword or a page boundary respectively. On Linux,
- * we use objtool to insert the archive into the bootloader, and objtool will
- * move the ELF section headers so they follow the cookie, so we need to search backward
- * past the section headers to find the cookie.
+ * we use objtool to insert the archive into the bootloader and on runtime parse the
+ * ELF headers to extract the precise end offset.
  */
 #if defined(WIN32)
 #define SEARCH_SIZE (8 + sizeof(COOKIE))
-#else
+#elif defined(__APPLE__)
```

Comment:
Why not special case Linux instead? This will also affect other platforms, like Solaris or AIX.

Reply:
AIX, HPUX, Solaris are all also ELF systems. Does PyInstaller embed the payload differently on other Unixes? The comment does not single out the other platforms.

I haven't looked into the embedding process up to now. On quick glance `binding/api.py` gives me the impression that data is appended to the end of the binary on non-linux Unixes. Only OSX gets the additional fixup call for the MachO binary. That would place the payload and cookie right at the end of the binary, which would render the above SEARCH_SIZE correct.

Reply:
I suppose other platforms which use ELF should work too. AIX uses XCOFF though, so it won't work there.

Reply:
I wonder whether there is some precompiler symbol telling if ELF is used as output format?